thirdweb/core/classes/ipfs_storage.py

```diff
@@ -0,0 +1,224 @@
+import re
+import json
+from requests import get, Response, post
+from typing import Any, Dict, List, TextIO, BinaryIO, Union, cast
+from thirdweb.common.error import FetchException, UploadException
+from thirdweb.constants.urls import (
+    DEFAULT_IPFS_GATEWAY,
+    PINATA_IPFS_URL,
+    TW_IPFS_SERVER_URL,
+)
+from thirdweb.core.helpers.storage import (
+    replace_file_properties_with_hashes,
+    replace_hash_with_gateway_url,
+    resolve_gateway_url,
+)
+from thirdweb.types.storage import CidWithFileName, UriWithMetadata
+
+
+class IpfsStorage:
```

Comment:
could we make sure that this inherits from an `ABC`? We need a strict interface for storage so its easily swappable.

https://docs.python.org/3/library/abc.html